src/tsdb.c

```diff
@@ -109,23 +129,24 @@ int SeriesIteratorGetNext(SeriesIterator *iterator, Sample *currentSample) {
         Chunk *currentChunk = iterator->currentChunk;
         if (ChunkGetLastTimestamp(currentChunk) < iterator->minTimestamp)
         {
-            iterator->currentChunk = currentChunk->nextChunk;
+            if (!RedisModule_DictNextC(iterator->dictIter, NULL, (void*)&iterator->currentChunk))
+                iterator->currentChunk = NULL;
             iterator->chunkIteratorInitialized = FALSE;
             continue;
         }
         else if (ChunkGetFirstTimestamp(currentChunk) > iterator->maxTimestamp)
         {
             break;
         }
-        
```

Comment:
why remove this whitespace? it allows the eyes to seperate between those two blocks of `if`